File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.py
class Solution(object):
    def findDuplicates(self, nums):
        ans = []
        for i in range(len(nums)):
            idx = abs(nums[i]) - 1
            nums[idx] *= -1
            if nums[idx] > 0:
                ans.append(abs(nums[i]))
        
        return ans
        

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
    let ans = []
    for(let i = 0; i<nums.length; i++) {
        let idx = Math.abs(nums[i]) - 1;
        nums[idx] *= -1;
        if(nums[idx] > 0) ans.push(Math.abs(nums[i]));
    }

    return ans;
};